dev/VSIX/ItemTemplates/Neutral/CppWinRT/TemplatedControl/TemplatedControl.cpp

```diff
@@ -17,16 +12,6 @@ namespace winrt::$rootnamespace$::implementation
 {
     $safeitemname$::$safeitemname$()
     {
-        DefaultStyleKey(winrt::box_value(L"$rootnamespace$.$safeitemname$"));
-    }
-
-    int32_t $safeitemname$::MyProperty()
-    {
-        throw hresult_not_implemented();
-    }
-
-    void $safeitemname$::MyProperty(int32_t /* value */)
-    {
-        throw hresult_not_implemented();
+        DefaultStyleKey(winrt::box_value(xaml_typename<class_type>()));
```

Comment:
>xaml_typename<class_type>() [](http://example.com/codeflow?start=41&length=27)

XAML accepts boxed strings so this change isn't necessary. Can you provide a min repro project for the crash you were encountering with the previous incarnation?

Reply:
It's not a crash, it's a compile error. C++/WinRT `hstring` does not implicitly convert to `IInspectable`. It needs to be explicitly done (via `winrt::box_value`). The template won't compile by default.

Reply:
That's correct. However, the change being proposed here is simply from `hstring` to `xaml_typename`; the enclosing `winrt::box_value()` call isn't being touched.

Reply:
The primary benefit of using `xaml_typename<class_type>()` over 'hstring' that I'm aware of is that it simplifies renaming the class. But the argument being made here instead is that there's an *error* associated with the `hstring` variant so that's why I want to dig deeper.

Reply:
You're right DefaultStyleKey can be set to a boxed hstring as well, it doesn't have to be a TypeName. As an alternative, winrt::name_of could be used instead of winrt::xaml_typename. There are a few differences between the name_of and xaml_typename, for example, name_of<Windows::Foundation::Point> returns "Windows.Foundation.Point" while xaml_typename<Windows::Foundation::Point> returns "Point". This doesn't matter in this case though.